include/seqan/basic/array_construct_destruct.h

```diff
@@ -251,7 +251,8 @@ struct ValueConstructor_
     {
         typedef typename Value<TIterator>::Type    TValue;
         typedef typename RemoveConst<TValue>::Type TNonConstValue;
-        new( (void*) & value(it) ) TNonConstValue(SEQAN_FORWARD(TParam, param_));
+        new( (void*) & value(it) ) TNonConstValue();
```

Comment:
std::string has no constructor for Segment<std::strng ...> but it does have an overloaded assign.